### PR TITLE
AKS - various fixes to edit mode

### LINKED ISSUE #10949: [v2.9] AKS: highest k8s version getting selected while editing cluster
**Setup**
Rancher version: v2.9-6d87a11ea46b7571646d7c3d7af704584c39fd62-head
Browser type & version: Chrome latest

**Describe the bug**
AKS (Vue UI): Highest k8s version getting selected while editing cluster.

**To Reproduce**
- Provision AKS cluster from Rancher with version less than highest supported version
- After cluster becomes Active, edit cluster and check Kubernetes version field
- At this step, highest version is selected by default

**Screenshots**
image

**Expected Result**
Current k8s version of cluster must be auto-selected while editing cluster, as in old UI.

### LINKED ISSUE #10950: [v2.9] AKS: Unable to add/upgrade Nodepool
**Setup**
Rancher version: _v2.9-6d87a11ea46b7571646d7c3d7af704584c39fd62-head_
Browser type & version: Chrome latest

**Describe the bug**
AKS (Vue UI): Unable to add new Nodepool or Upgrade k8s version of existing Nodepool
It seems _orchestratorVersion_ is not set in API payload.

**To Reproduce**
1. Provision AKS cluster from Rancher UI with version less than highest supported version
2. After cluster becomes Active, edit cluster and upgrade cluster k8s version
3. At this step, there is no field on UI to update Nodepool k8s version
4. After step 1, edit the cluster and add new Nodepool, operation does not completes.

**Screenshots**
No Nodepool k8s version:
image

Adding new Nodepool:
image

**Expected Result**
- There must be UI field on UI for Nodepool k8s version as on old UI
- Adding new Nodepool must complete successfully (operation works fine via API)

### LINKED ISSUE #10952: [v2.9] AKS: Labels/Taints not visible for Nodepool
**Setup**
Rancher version: `v2.9-6d87a11ea46b7571646d7c3d7af704584c39fd62-head`
Browser type & version: Chrome latest

**Describe the bug**
AKS (Vue UI): Labels/Taints not visible for Nodepool 

**To Reproduce**
- Provision AKS cluster from Rancher UI
- For Nodepool, add labels/taints while provisioning
- Wait for Cluster to become Active

API payload sent:
```
    "nodePools": [
      {
        "availabilityZones": [
          "1",
          "2",
          "3"
        ],
        "count": 1,
        "enableAutoScaling": false,
        "maxPods": 110,
        "maxSurge": "1",
        "mode": "System",
        "name": "agentpool",
        "nodeLabels": {
          
        },
        "nodeTaints": [
          "k1=v1:PreferNoSchedule"
        ],
        "orchestratorVersion": "",
        "osDiskSizeGB": 128,
        "osDiskType": "Managed",
        "osType": "Linux",
        "vmSize": "Standard_DS2_v2",
        "labels": {
          "k2": "v2"
        }
      }
    ],
```

**Screenshots**
After cluster is provisioned:
image

**Expected Result**
Labels/Taints for Nodepool must be visible on UI

### LINKED ISSUE #10960: [v2.9] AKS: Unable to change Nodepool mode
**Setup**
Rancher version: v2.9-6d87a11ea46b7571646d7c3d7af704584c39fd62-head
Browser type & version: Chrome latest

**Describe the bug**
AKS (Vue UI): Unable to change Nodepool mode, from System to User and vice-versa.

**To Reproduce**
- Provision AKS cluster from Rancher UI with two Nodepools and different modes

**Screenshots**
image

**Expected Result**
User must be able to change Nodepool mode, as in old UI.

### PR DESCRIPTION
**Pull request by mantis-toboggan-md**
### Summary
Fixes #10949 
Fixes #10950 
Fixes #10952 
Fixes #10960 

This PR also removes diffUpstreamSpec code from AKS utils, as it was moved to shell/dashboard to be used in all hosted providers.

### Occurred changes and/or fixed issues
#10950 - node pools should have a field to set `orchestratorVersion`. I have duplicated functionality observed in the Ember UI, which is described in the unit tests I've added around this input. 

#10952 - in addition to fixing and adding tests for the bug in the issue, when editing clusters through the azure portal I found that both key and value are required fields in each taint. I've added validation to verify that they're defined.

### Technical notes summary
In node pool taints, if the user enters a value but not a key, the value input shows a validation error icon/message, and vice versa. This was simpler than altering the form validation mixin to display error messages when an input has not yet been touched (I am reluctant to expand the functionality of a mixin we're in the process of replacing).

### Areas or cases that should be tested
Creating new clusters/adding new node pools: 
 - taint key and value are both required
 - node pool version display should be automatically updated when the cluster version is changed
 
Editing clusters/existing node pools:
- if a new k8s version is selected, node pools should have an info banner about how to upgrade node pools
- if the cluster k8s version is greater than a node pool's version and the cluster version is not being upgraded, there should be an option to upgrade the node pool.

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone 
- [x] The PR template has been filled out
- [x] The PR has been self reviewed 
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

### CODE REVIEW

File: pkg/aks/components/AksNodePool.vue [hunk 1]
```diff
@@ -107,7 +118,33 @@ export default defineComponent({
...
+      return this.originalClusterVersion !== this.clusterVersion;
+    },
+
+    // offer a k8s version upgrade if the node pool is not on the same version as the cluster and the cluster is not currently being upgraded
+    upgradeAvailable(): boolean {
+      if (this.mode === _CREATE || this.pool._isNewOrUnprovisioned) {
+        return false;
+      }
+
+      return this.clusterVersion !== this.originalOrchestratorVersion;
```
Comment (momesgin) [inline pkg/aks/components/AksNodePool.vue:133]: I think we should also check for falsy values like `undefined`, for example if you go to edit mode while the cluster is still in progress we'll show this:
image

Comment (mantis-toboggan-md) [reply pkg/aks/components/AksNodePool.vue]: Interesting, I haven't been able to reproduce this bug myself. In this scenario, the cluster/agentpool were created with version 1.29.2, then, upon opening the edit form, you see this incorrect checkbox? It's easy enough to account for the possibility that `originalOrchestratorVersion` is undefined but I'm not sure about testing, as I can't repro

Comment (momesgin) [reply pkg/aks/components/AksNodePool.vue]: I re-tested with your latest changes and it's not reproducible on my end either.

**Review by momesgin — APPROVED** [review]
LGTM 🚀

**Pull request merged**